upload/uploader.go

```diff
@@ -86,7 +88,25 @@ func (u Uploader) Upload() error {
 		return errors.WithStack(err)
 	}
 
-	return u.SendBatches(testReport, batchLinks.Links.PostBatch)
+	postBatchURL := u.TransformPostBatchURL(batchLinks.Links.PostBatch)
+
+	return u.SendBatches(testReport, postBatchURL)
+}
+
+func (u Uploader) TransformPostBatchURL(rawUrl string) string {
+	parsed, err := url.Parse(rawUrl)
+
+	if err != nil {
+		return rawUrl
```

Comment:
If this is an unusable value, I guess the reporter will fail further downstream at the http post request?

Reply:
In case it wasn't obvious, I'm new to Go. So this might not be desirable. 😄 Would you recommend a `panic` instead?

Reply:
I'm not familar with all of the error handling in place but it looks like you could do something like this if you wanted to handle the url parsing error directly:

```golang
func (u Uploader) TransformPostBatchURL(rawUrl string) (string, errror) {
  parsed, err := url.Parse(rawUrl)

  if err != nil {
    return nil, err
  }
  // snip
}
```

and change the calling code to:
```golang
postBatchURL, err := u.TransformPostBatchURL(batchLinks.Links.PostBatch)

if err != nil {
  return errors.WithStack(err)
}

return u.SendBatches(testReport, postBatchURL)
```

Reply:
FWIW i thought about this but having an unusable URL it's extremely unlikely and I think this a nice to have but not super necessary. Personally I find super annoying those `if` 🙈

Reply:
Ah, OK!

It's unlikely, @ale7714, but @dblandin's suggestion means it'll be much easier to discover and debug.